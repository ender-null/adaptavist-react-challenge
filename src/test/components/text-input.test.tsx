import React from "react";
import { fireEvent, render } from "@testing-library/react";
import TextInput, { TextInputProps } from "../../components/text-input";

describe("TextInput", () => {
  const props: TextInputProps = {
    text: "test text",
    placeholder: "Placeholder",
    updateText: (text: string) => {
      return text;
    },
  };
  test("should render the component", () => {
    const { asFragment } = render(<TextInput {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("should run call updateText with a delay when changing text", async () => {
    const { getByPlaceholderText } = render(<TextInput {...props} />);
    const input = getByPlaceholderText("Placeholder");
    if (input) {
      fireEvent.change(input, { target: { value: "New text" } });
      await new Promise((r) => setTimeout(r, 500));
      expect(input.textContent).toBe("New text");
    }
  });
});
