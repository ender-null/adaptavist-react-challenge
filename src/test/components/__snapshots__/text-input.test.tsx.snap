// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render the component 1`] = `
<DocumentFragment>
  <textarea
    class="textarea"
    placeholder="Placeholder"
  >
    test text
  </textarea>
</DocumentFragment>
`;
